### PR TITLE
Seed Extractor stores the extracted seeds(On Right Click) in the machine rather than spewing them out on the turf

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
Say you attack the seed extractor with a pumpkin, and it extracts 4 pumpkin seeds. Rather than dumping those 4 seed packets on the turf it will now store them in the machine.

This feature is optional only if you right click. Left click does the normal stuff

## Why It's Good For The Game
Saves you the manual work of picking up those packets from the turf and again putting them in the machine. Keeps the turf clean.

## Changelog
:cl:
qol: right clicking the seed extractor with a plant/food stores the extracted seeds in the machine.
/:cl:

### CODE REVIEW

**Comment by SyncIt21** [comment]
> I don't think I like this, if I want a seed from a plant I don't want to have to extract it, open up the UI, then search through hundreds of seeds to find the exact sample I just took. I just want to pick it up off the top.

I mean there is a search bar in the UI to find what your looking for. 
Maybe I make it optional then say if I right click on the machine with the plant then I store it in the machine but if I left click then it does the normal stuff?

**Comment by SyncIt21** [comment]
> That sounds preferable, just make sure it comes with screentips and the like

I'll do that then

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/seed_extractor.dm [hunk 1]
```diff
@@ -156,6 +162,10 @@
 
 	return ..()
 
+/obj/machinery/seed_extractor/attackby_secondary(obj/item/weapon, mob/user, params)
+	. = ..()
+
```
[the remark this answers is not in the document]
Comment (SyncIt21) [reply code/modules/hydroponics/seed_extractor.dm:167]: Ohh right I forgot to remove this after i learnt about  click params. Thanks for pointing it out

File: code/modules/hydroponics/seed_extractor.dm [hunk 2]
```diff
@@ -136,6 +137,11 @@
 		return TRUE
 
 	if(seedify(attacking_item, -1, src, user))
+		if(LAZYACCESS(params2list(params), RIGHT_CLICK))
+			//find all seeds lying on the turf and add them to the machine
+			var/obj/item/seeds/to_store
+			while((to_store = locate(/obj/item/seeds, loc)) != null)
+				add_seed(to_store, null)
```
[the remark this answers is not in the document]
Comment (SyncIt21) [reply code/modules/hydroponics/seed_extractor.dm:144]: It's an added bonus that is picks up all seeds.  But yeah I'll try to filter them out using seedify

**Comment by SyncIt21** [comment]
There is a bug here. max_seeds doesn't limit the number of seeds you can store here. It only checks against how many seeds you are trying to insert but it doesn't check against how many seeds are actually in the machine. I'm gonna fix this

**Comment by SyncIt21** [comment]
Done

**Comment by SyncIt21** [comment]
I have no pending work on this. Ready for Merge/Review

**Comment by ExcessiveUseOfCobblestone** [comment]
why WOULDNT you want this as baseline? If its that pressing you could just put the first seed into your hand and the rest go thru UI.

If we are talking clicks having to manually put all seeds into the extractor is more clicks than what it would take to pull out those if they would go into the machine, not to mention the location of where you clicking doesnt change on the UI unlike ground > extractor for storage.

**Comment by SyncIt21** [comment]
> why WOULDNT you want this as baseline? If its that pressing you could just put the first seed into your hand and the rest go thru UI.
> 
> If we are talking clicks having to manually put all seeds into the extractor is more clicks than what it would take to pull out those if they would go into the machine, not to mention the location of where you clicking doesnt change on the UI unlike ground > extractor for storage.
> 
> Maybe im blessed by RNG but I always get multiple seeds, which is usually a pain to put them in to clean up the area (I usually just resort to portable for this reason anyways).

We can make both sides happy.  Right clicking on the extractor will put all seeds into the machine and wont cause a mess and left click is for everyone else who doesn't care about the mess.
Making it a baseline would have caused others to complain so might as well take this route

**Comment by SyncIt21** [comment]
Merge/Review anyone? I have no pending work on this

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Two small things, otherwise looks good

**Comment by SyncIt21** [comment]
Done

**Pull request merged**